### DISCUSSION TITLE
Can pytorch-accelerated be used in other types of tasks, such as detection, reconstruction?

### DISCUSSION THREAD

**Discussion by cekxm** [Chris-hughes10/pytorch-accelerated#21] [Q&A] [answered]
pytorch-accelerated is great. However, how can we adapt it to tasks other than classification?

**Comment by Chris-hughes10** [comment, marked as answer]
Hi @cekxm, thanks for the kind words! PyTorch-accelerated can be used for lots of tasks other than classification, for example, [this blog post](https://medium.com/p/8e3cd3ec8bd8 ) demonstrates using it for a recommendation problem and I have recently been using it to train Yolox for object detection (blog post hopefully coming soon)! 

In many cases, you can get away with using the trainer directly as long as your dataset returns (xb, yb) that are passed to a loss function. In cases where that doesn't work, you will have to subclass the trainer, such as [this example](https://github.com/Chris-hughes10/pytorch-accelerated/blob/main/examples/nlp/hf_bert_glue_mrpc.py).

Were there any tasks that you had in mind?

**Reply by cekxm** [reply]
Hi @Chris-hughes10 , thanks for the reply. I now realized that PyTorch-accelerated is widely applicable. Currently, we mainly focus on low-level image restoration and pedestrian/ship detection. So looking forward to read your post on Yolox for object detection.
